app/src/main/java/org/oppia/android/app/application/AbstractOppiaApplication.kt

```diff
@@ -47,6 +51,16 @@ abstract class AbstractOppiaApplication(
     // TODO(#4751): Re-enable WorkManager for S+.
     if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) {
       FirebaseApp.initializeApp(applicationContext)
+      val isDebugBuild = 0 != applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE
+      if (isDebugBuild) {
```

Comment:
This feels like something we could decide based on the type of application being compiled (e.g. dev build always having this on, but others not). When do we need to use debug vs. non-debug app check?

Reply:
Release Appcheck will be used by release builds e.g. alpha, beta and GA but Debug is only used in dev build for now, and in future if we implement appcheck in test environments and CI.

Reply:
I switched this check for a build flavor check. PTAL.

Reply:
Got it that makes sense--thanks!